### PR TITLE
Feat/wallet: NEP-2 Storage Support

### PR DESCRIPTION
**Pull request by phetter**
- Configuration updates
- Readme updates
- src/nodejs/account/getNep2EncryptedKey.js
- account.getNep2EncryptedKey()

### CODE REVIEW

File: src/nodejs/account/account.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ exports.getDefaultAccount = () => {
 
   var account = _.findWhere(accounts, {default: true})
 
-  if(account && account.path !== null) {
+  if (account && account.path !== null) {
```
Comment (DalderupMaurice) [inline src/nodejs/account/account.js]: the !== null part is not needed

File: src/nodejs/account/account.js [hunk 2]
```diff
@@ -86,3 +88,19 @@ exports.list = (configData, accountName) => {
...
+
+
+// Return NEP-2 encrypted key from account config data.
+// This searches the data for the account with accountName and plucks that key.
+
+exports.getNep2EncryptedKey = (configData, accountName) => {
+  var accounts = this.list(configData)
+  var account
+
+  if (accountName) account = accounts[accountName]
```
Comment (DalderupMaurice) [inline src/nodejs/account/account.js]: how are you always sure that this index exists?

File: src/nodejs/account/account.js [hunk 3]
```diff
@@ -86,3 +88,19 @@ exports.list = (configData, accountName) => {
...
+  else account = _.findWhere(accounts, {default: true})
+
+  if (account && account.nep2EncryptedKey) return account.nep2EncryptedKey
+
+  else return null
```
Comment (DalderupMaurice) [inline src/nodejs/account/account.js]: do we want to return null here?

File: src/nodejs/account/cli/getNep2EncryptedKey.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  path = program.config
+  configData = cfg.load(path)
+} else configData = config
+
+if (program.Debug) {
+  print('DEBUGGING');
+}
+
+if (program.watch) {
+  var accounts = account.get_watch_addresses(configData)
```
Comment (DalderupMaurice) [inline src/nodejs/account/cli/getNep2EncryptedKey.js]: wasn't everything moved to camelCasing?

File: src/nodejs/account/cli/getNep2EncryptedKey.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  if (accounts && accounts.length) {
+    print('result:')
+    accounts.forEach((o) => {
+      dbg.logDeep('', o)
```
Comment (DalderupMaurice) [inline src/nodejs/account/cli/getNep2EncryptedKey.js]: we're only printing? No other actions required?

File: src/nodejs/account/cli/list.js [hunk 6]
```diff
@@ -42,25 +42,27 @@ if (program.Debug) {
 
 
 if (program.watch) {
-  var result = account.get_watch_addresses(configData)
+  var accounts = account.get_watch_addresses(configData)
```
Comment (DalderupMaurice) [inline src/nodejs/account/cli/list.js]: same here - camelCasing?

File: src/nodejs/account/cli/list.js [hunk 7]
```diff
@@ -42,25 +42,27 @@ if (program.Debug) {
...
 
-  if (result && result.length)
-  print('result:')
-  result.forEach((r) => {
-    dbg.logDeep('', r)
-  })
+  if (accounts && accounts.length) {
+    print('result:')
+    accounts.forEach((o) => {
+      dbg.logDeep('', o)
```
Comment (DalderupMaurice) [inline src/nodejs/account/cli/list.js]: same comment :P

File: src/nodejs/account/cli/list.js [hunk 8]
```diff
@@ -42,25 +42,27 @@ if (program.Debug) {
...
-
-  if (result) {
+  var result
+  var accounts = account.list(configData)
 
+  if (accounts) {
     if (program.name) {
-      json.findAllKeysWhere(result, program.name, (key, val) => {
+      json.findAllKeysWhere(accounts, program.name, (key, val) => {
         result = val
```
Comment (DalderupMaurice) [inline src/nodejs/account/cli/list.js]: you're overriding this variable with each loop? This way  only the last loop is stored and all other iterations are lost

File: README.md [hunk 9]
```diff
@@ -303,16 +306,28 @@ The path item of each config entry points to a json config file that adheres to
 {
   "accounts": {
     "one": {
+      "type": "neo",
```
[the remark this answers is not in the document]
Comment (phetter) [reply README.md]: Yes.

File: src/nodejs/account/cli/list.js (hunk 8 above)
Comment (phetter) [reply src/nodejs/account/cli/list.js]: I'm okay with this for now as there should only be one account by that name. I'll add a note though.

**Pull request merged**